Sources/Subprocess/Platforms/Subprocess+Windows.swift

```diff
@@ -629,6 +750,148 @@ extension Executable {
             return executablePath.string
         }
     }
+
+    /// `CreateProcessW` allows users to specify the executable path via
+    /// `lpApplicationName` or via `lpCommandLine`. However, only `lpCommandLine` supports
+    /// path searching, whereas `lpApplicationName` does not. In order to support the
+    /// "argument 0 override" feature, Subprocess must use `lpApplicationName` instead of
+    /// relying on `lpCommandLine` (so we can potentially set a different value for `lpCommandLine`).
+    ///
+    /// This method replicates the executable searching behavior of `CreateProcessW`'s
+    /// `lpCommandLine`. Specifically, it follows the steps listed in `CreateProcessW`'s documentation:
+    ///
+    /// 1. The directory from which the application loaded.
+    /// 2. The current directory for the parent process.
+    /// 3. The 32-bit Windows system directory.
+    /// 4. The 16-bit Windows system directory.
+    /// 5. The Windows directory.
+    /// 6. The directories that are listed in the PATH environment variable.
+    ///
+    /// For more info:
+    /// https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessw
+    internal func possibleExecutablePaths(
+        withPathValue pathValue: String?
+    ) -> Set<String> {
+        func insertExecutableAddingExtension(
+            _ name: String,
+            currentPath: String,
+            pathExtensions: Set<String>,
+            storage: inout Set<String>
+        ) {
+            let fullPath = FilePath(currentPath).appending(name)
+            if !name.hasExtension() {
+                for ext in pathExtensions {
+                    var path = fullPath
+                    path.extension = ext
+                    storage.insert(path.string)
+                }
+            } else {
+                storage.insert(fullPath.string)
+            }
+        }
+
+        switch self.storage {
+        case .executable(let name):
+            var possiblePaths: Set<String> = []
+            let currentEnvironmentValues = Environment.currentEnvironmentValues()
+            // If `name` does not include extensions, we need to try these extensions
+            var pathExtensions: Set<String> = Set(["com", "exe", "cmd", "bat"])
```

Comment:
why hardcode these extensions at all? why not rely solely on PATHEXT? also, the default order in PATHEXT is com,exe,bat,cmd, rather than cmd appearing before bat